### PR TITLE
🎦 Move SD provider utils function to utils folder

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes

1. Move SD provider utils function to utils folder
2. Fix the wrong link in sd readme.md.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: examples/stable_diffusion/README.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ Refer to the corresponding section in the DirectML READMEs for more details on t
 - [Stable Diffusion XL](../directml/stable_diffusion_xl/README.md#test-inference)
 
 
-## Stable Diffusion Optimization  with OpenVINO
+## Stable Diffusion Optimization with OpenVINO
```
Comment (trajepl) [inline examples/stable_diffusion/README.md:123]: The extra blank whitespace will lead the wrong link error from above tag link.

File: examples/stable_diffusion/stable_diffusion.py
Comment (jambayk) [inline examples/stable_diffusion/stable_diffusion.py:1]: Is this change just to clean up the code? Or in preparation to add extra utils or provider support?

Comment (trajepl) [reply examples/stable_diffusion/stable_diffusion.py:1]: Yes, for add qnn support.

**Review by jambayk — APPROVED** [review]

File: examples/stable_diffusion/stable_diffusion.py
Comment (jambayk) [reply examples/stable_diffusion/stable_diffusion.py:1]: I see. thanks! approved.

**Pull request merged**